dp/extract_link_from_pages.py

```diff
@@ -125,9 +125,10 @@ def extract_links(dump_prefix, out, encoding, normalizer):
     all_files = []
     for directory in os.listdir(path=dump_prefix_abs):
         subdir = os.path.join(dump_prefix_abs, directory)
-        for f in os.listdir(path=subdir):
-            full_path = os.path.join(subdir, f)
-            all_files.append(full_path)
+        if os.path.isdir(subdir):
```

Comment:
I do not understand why this is needed. Maybe you need to add a comment?

Reply:
I don't understand the logic in the code (definitely needs a lot of comments explaining what is happening).
Also, maybe we should use  a logger and log hyperlinks we are not able to resolve. This usually helps a great deal in finding edge cases. @shyamupa does that make sense?

Reply:
What happened on my Mac is that it will automatically create a .DS_store file in the folder. If I did not add this if statement, it will try to open the .DS_store and this will throw an error.

Reply:
You can add something int he make file that recursively removes the .DS_Store folder. 
Also, my comment about code-comments and logging was in general about the whole file and not just this segment.